File: Arrays/Validate Subsequence/python/solution2.py

# O(n) time complexity

def isValidSubsequence(array, sequence):
        # Write your code here.
        seqID = 0
        for num in array:
            if seqID == len(sequence):
                break
            if num == sequence[seqID]:
                seqID+=1
        return seqID == len(sequence)

File: old/Arrays-Old/Validate Subsequence/c++/solution2.cpp
#include <vector>
using namespace std;

bool isValidSubsequence(vector<int> array, vector<int> sequence) {
  // Write your code here.
	int arid = 0;
	int sid =0;
	for (auto value : array) {
		if (sid == sequence.size())
			break;
        if (value == sequence[sid])
            sid+=1;
	}
  return sid == sequence.size();
}
